pytorch_lightning/utilities/cli.py

```diff
@@ -260,6 +337,62 @@ def instantiate_trainer(self) -> None:
             self.config_init['trainer']['callbacks'].append(config_callback)
         self.trainer = self.trainer_class(**self.config_init['trainer'])
 
+    def add_configure_optimizers_method_to_model(self) -> None:
+        """Adds to the model an automatically generated configure_optimizers method
+
+        If a single optimizer and optionally a scheduler argument groups are added to the parser as 'AUTOMATIC',
+        then a `configure_optimizers` method is automatically implemented in the model class.
+        """
+
+        def get_automatic(class_type: Union[Type, Tuple[Type, ...]]) -> List[str]:
+            automatic = []
+            for key, (base_class, link_to) in self.parser.optimizers_and_lr_schedulers.items():
+                if not isinstance(base_class, tuple):
+                    base_class = (base_class, )
+                if link_to == 'AUTOMATIC' and any(issubclass(c, class_type) for c in base_class):
+                    automatic.append(key)
+            return automatic
+
+        optimizers = get_automatic(Optimizer)
+        lr_schedulers = get_automatic(LRScheduler)
+
+        if len(optimizers) == 0:
+            return
+
+        if len(optimizers) > 1 or len(lr_schedulers) > 1:
+            raise RuntimeError(
+                f"`{self.__class__.__name__}.add_configure_optimizers_method_to_model` expects at most one optimizer "
+                f"and one lr_scheduler to be 'AUTOMATIC', but found {optimizers+lr_schedulers}."
+            )
+
+        if type(self.model).configure_optimizers != LightningModule.configure_optimizers:
```

Comment:
not sure but one could probably use the utility function in Lightning `is_overridden `

Reply:
Good to know. Will try to change it to use `is_overridden`.

Reply:
Done